recipes/libnice/all/conanfile.py

```diff
@@ -0,0 +1,109 @@
+import os
+from conan import ConanFile
+from conan.tools.meson import Meson, MesonToolchain
+from conan.tools.files import copy, get, rmdir, rename, chdir, replace_in_file
+from conan.tools.layout import basic_layout
+from conan.tools.gnu import PkgConfigDeps
+from conan.errors import ConanInvalidConfiguration
+
+
+class LibniceConan(ConanFile):
+    name = "libnice"
+    version = "0.1.19"
+    homepage = "https://libnice.freedesktop.org/"
+    license = "MPL-1.1 AND LGPL-2.1-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "a GLib ICE implementation"
+    topics = ("ice", "stun", "turn")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "crypto_library": ["openssl", "win32"],
+        "with_gstreamer": [True, False],
+        "with_gtk_doc": [True, False],
+        "with_introspection": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_gstreamer": False,
+        "with_gtk_doc": False,
+        "with_introspection": False}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.crypto_library = "win32"
+        else:
+            self.options.crypto_library = "openssl"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.os != "Windows" and self.options.crypto_library == "win32":
+            raise ConanInvalidConfiguration(
+                "crypto_library=win32 is not supported on non-Windows")
+        if self.settings.os == "Windows" and self.options.with_gtk_doc:
+            raise ConanInvalidConfiguration(
+                "gtk-doc is disabled by libnice while building on Windows")
+
+    def requirements(self):
+        self.requires("glib/2.75.0")
+        if self.options.crypto_library == "openssl":
+            self.requires("openssl/1.1.1s")
+        if self.options.with_gstreamer:
+            self.requires("gstreamer/1.19.2")
+
+    def build_requirements(self):
+        self.build_requires("meson/0.64.1")
+        self.build_requires("pkgconf/1.9.3")
+        self.build_requires("glib/2.75.0") # for glib-mkenums
+        if self.options.with_introspection:
+            self.build_requires("gobject-introspection/1.72.0")
```

Comment:
```suggestion
        self.tool_requires("meson/0.64.1")
        self.tool_requires("pkgconf/1.9.3")
        self.tool_requires("glib/2.75.0") # for glib-mkenums
        if self.options.with_introspection:
            self.tool_requires("gobject-introspection/1.72.0")
```
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#requirements